src/lib/datepicker/calendar.ts

```diff
@@ -133,28 +137,46 @@ export class MatCalendar<D> implements AfterContentInit, OnDestroy, OnChanges {
   private _clampedActiveDate: D;
 
   /** Whether the calendar is in month view. */
-  _monthView: boolean;
+  _currentView: 'month' | 'year' | 'multi-year';
 
   /** The label for the current calendar view. */
   get _periodButtonText(): string {
-    return this._monthView ?
-        this._dateAdapter.format(this._activeDate, this._dateFormats.display.monthYearLabel)
-            .toLocaleUpperCase() :
-        this._dateAdapter.getYearName(this._activeDate);
+    if (this._currentView == 'month') {
+      return this._dateAdapter.format(this._activeDate, this._dateFormats.display.monthYearLabel)
+          .toLocaleUpperCase();
+    }
+    if (this._currentView == 'year') {
+      return this._dateAdapter.getYearName(this._activeDate);
+    }
+    let curYear = this._dateAdapter.getYear(this._activeDate);
+    let firstYear = this._dateAdapter.getYearName(
+        this._dateAdapter.createDate(curYear - curYear % 24, 0, 1));
+    let lastYear = this._dateAdapter.getYearName(
+        this._dateAdapter.createDate(curYear + yearsPerPage - 1 - curYear % 24, 0, 1));
+    return `${firstYear} \u2013 ${lastYear}`;
```

Comment:
Would this need to be internationalized?

Reply:
It probably should be. I don't think any of our `*Intl` objects currently have placeholders, so we'd have to make up some convention for that.

Should we switch to using the official Angular i18n stuff? It seems weird that we're just rolling our own.